File: app/src/main/java/fr/newglace/notedesnazes/Styles/Dialog/QRCode.java
package fr.newglace.notedesnazes.Styles.Dialog;

import android.app.Activity;
import android.app.Dialog;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.util.DisplayMetrics;
import android.widget.ImageView;

import androidx.annotation.RequiresApi;

import fr.newglace.notedesnazes.R;
import fr.newglace.notedesnazes.Styles.Colors;
import fr.newglace.notedesnazes.Styles.reSize2;

public class QRCode extends Dialog {
    private ImageView image;

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    public QRCode(Activity activity) {
        super(activity, R.style.Theme_AppCompat_DayNight_Dialog);
        setContentView(R.layout.dialog_qrcode_image);

        image = findViewById(R.id.qr_code);
        ImageView imageView = findViewById(R.id.imageView);
        reSize2 size = new reSize2();

        final DisplayMetrics metrics = new DisplayMetrics();
        activity.getWindowManager().getDefaultDisplay().getMetrics(metrics);

        int phoneWidth = metrics.widthPixels;
        int phoneHeight = metrics.heightPixels;
        int qrCodeValue = Math.min((int) (phoneWidth / 2d), (int) (phoneHeight / 3d));
        size.reSize2(imageView, (int) (qrCodeValue * 1.2d), (int) (qrCodeValue * 1.2d));
        size.reSize2(image, qrCodeValue, qrCodeValue);

        Colors colors = new Colors();
        Drawable draw2 = activity.getDrawable(R.drawable.bg_notes);
        colors.editColor(imageView, 0, draw2);
    }

    public ImageView getImage() {
        return image;
    }

    public void build() {
        show();
    }
}

File: app/src/main/java/fr/newglace/notedesnazes/styles/dialog/QRCode.kt
package fr.newglace.notedesnazes.styles.dialog

import android.app.Activity
import android.app.Dialog
import android.os.Build
import android.util.DisplayMetrics
import android.widget.ImageView
import androidx.annotation.RequiresApi
import fr.newglace.notedesnazes.R
import fr.newglace.notedesnazes.styles.Colors
import fr.newglace.notedesnazes.styles.ReSize

class QRCode @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP) constructor(activity: Activity) : Dialog(activity, R.style.Theme_AppCompat_DayNight_Dialog) {
    val image: ImageView

    fun build() {
        show()
    }

    init {
        setContentView(R.layout.dialog_qrcode_image)
        image = findViewById(R.id.qr_code)
        val imageView = findViewById<ImageView>(R.id.imageView)
        val size = ReSize()
        val metrics = DisplayMetrics()
        activity.windowManager.defaultDisplay.getMetrics(metrics)
        val phoneWidth = metrics.widthPixels
        val phoneHeight = metrics.heightPixels
        val qrCodeValue = Math.min((phoneWidth / 2.0).toInt(), (phoneHeight / 3.0).toInt())
        size.reSizing(imageView, (qrCodeValue * 1.2).toInt(), (qrCodeValue * 1.2).toInt())
        size.reSizing(image, qrCodeValue, qrCodeValue)
        val colors = Colors(activity)
        val draw2 = activity.getDrawable(R.drawable.bg_notes)
        colors.editColor(imageView, 0, draw2)
    }
}